Mathlib/Algebra/Order/BigOperators/GroupWithZero/List.lean

```diff
@@ -58,6 +58,23 @@ theorem prod_map_le_prod_map₀ {ι : Type*} {s : List ι} (f : ι → R) (g : 
       simp [ha]
     · apply (h0 _ _).trans (h _ _) <;> simp
 
+theorem prod_map_le_pow_lenght {F L : Type*} [MulPosMono R] [FunLike F L R] {f : F} {r : R}
+    {t : List L} (hf0 : ∀ x : L, 0 ≤ f x) (hf : ∀ x : R, x ∈ List.map f t → x ≤ r) :
```

Comment:
For `hf`, it might be easier to apply if you instead require `\forall x \in t, f x <= r`. Similarly for `hf0`, we only need it for `x \in t` I think.

Reply:
I think this has not been addressed? Or do you think the current statement is better?